include/common-types.h

```diff
@@ -0,0 +1,25 @@
+#ifndef CONNECT_SDK_COMMON_TYPES_H
+#define CONNECT_SDK_COMMON_TYPES_H
+
+#include <string>
+#include <vector>
+#include <map>
+#include <boost/move/unique_ptr.hpp>
+#include <boost/chrono/system_clocks.hpp>
+
+namespace prism {
+namespace connect {
+
+using std::string;
+using std::vector;
+using std::map;
+
+using boost::movelib::unique_ptr;
```

Comment:
template<typename T>
struct vector 
{ 
    typedef std::vector<T> t; 
};

Then you use it as
vector::t<T> vec;


There is more discussion on it here:
http://stackoverflow.com/questions/6907194/how-to-typedef-a-template-class

Some boost types go to std:: namespace in new compilers.
So, doing this using thing can cause issues with new compilers
if somebody will do using for both: std and prism::connect (or boost::).

My suggestion would be.
For std::vector and std::map to use fully qualified types with std.
It is not that big of hassle.
If we want to be able to switch to our type later ,then we shall do vector<>::t thing.

For boost::movelib::unique_ptr I would  do unique_ptr<>::t way.
Otherwise with std::unique_ptr in new compilers is just too easy to get a conflict
when asking to using namespace for both namespaces.